content/en/_index.md

```diff
@@ -1028,9 +1203,86 @@ def read_a_file file_path:
   puts "Contents: #{content}"
 end
 {{< /highlight >}}
+{{< /content-tab >}}
+{{< content-tab header="Terraform" >}}
+#### No CLIs
+
+Do not include CLI commands (such as gcloud or kubectl) inside
+of your sample via the `null_resource` resource.
+
+#### No Variables
+
+Don't include Terraform variables (`var.VARIABLE_NAME`). Variables
+enabling users to use Terraform samples not as snippets but as direct
+module references. Disabling the avenue of customization 
+reduces the risk so that users don't take accidental 
+dependency.
+Instead, hard code the resource names and attribute values.
+In some cases, [locals](#locals) are allowed.
+
+#### Locals {#locals}
+
+Sometimes multiple resources have a common string that is not
+accessible as a reference or there is no child/parent
+reference. Locals are allowed and encouraged for repeated text
+that a user may want to change. When a name is used multiple
+times, it's easier to copy, paste, and edit a local than search
+and replace the whole file. Locals are allowed and variable blocks
+are not because locals are an internal implementation and not
+part of the API  exposed via Terraform. Locals should be used only
+when necessary, and are best used when a repeated value is used
+throughout the configuration (3 or more times).
```

Comment:
suggestion: We try to keep guidelines concise and actionable. Could we edit this down, reduce the amount of justification text, and add an example?

Here's an example of reducing a first paragraph to the core concept a developer needs to know:

```suggestion
Use local variables to reuse a common string 3 or more times that is not
accessible as a resource reference.
```

We could add a second paragraph with more insight if required.


#StyleGuideFeedback

Reply:
Done